xdsl/dialects/csl.py

```diff
@@ -883,6 +892,50 @@ def verify_(self) -> None:
             raise VerifyException(f"{self.name} can only operate on mem1d_dsd type")
 
 
+FunctionSignatures = list[
+    tuple[AttrConstraint | Attribute | type[Attribute] | TypeVar, ...]
+]
+
+
+class _BuiltinDsdOpBase(IRDLOperation, ABC):
+    ops = var_operand_def()
+
+    @staticmethod
+    @abstractmethod
+    def get_signatures() -> FunctionSignatures:
+        raise NotImplementedError()
```

Comment:
I think you should be able to just do `def get_signatures() -> FunctionSignatures: ...`